rero_ils/modules/libraries/api_views.py

```diff
@@ -0,0 +1,71 @@
+# -*- coding: utf-8 -*-
+#
+# RERO ILS
+# Copyright (C) 2021 RERO
+# Copyright (C) 2021 UCLouvain
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, version 3 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Blueprint used for libraries API."""
+
+from __future__ import absolute_import, print_function
+
+from datetime import datetime, timedelta
+
+from flask import Blueprint, abort, jsonify, request
+
+from rero_ils.modules.decorators import check_logged_as_librarian
+from rero_ils.modules.libraries.api import Library
+from rero_ils.modules.utils import add_years, date_string_to_utc
+
+api_blueprint = Blueprint(
+    'api_library',
+    __name__,
+    url_prefix='/library'
+)
+
+
+@api_blueprint.route('/<library_pid>/closed_dates', methods=['GET'])
+@check_logged_as_librarian
+def list_closed_dates(library_pid):
+    """HTTP GET request to get the closed date about a library."""
+    library = Library.get_record_by_pid(library_pid)
+    if not library:
+        abort(404)
+
+    # get start date from 'from' parameter from query string request
+    start_date = request.args.get('from', datetime.now() - timedelta(days=31))
+    if isinstance(start_date, str):
+        start_date = date_string_to_utc(start_date)
+    start_date = start_date.replace(tzinfo=library.get_timezone())
+    # get end date from 'until' parameter from query string request
+    end_date = request.args.get('until', add_years(datetime.now(), 1))
```

Comment:
does this mean you return the closed dates of one complete year if the `until` is not given?

Reply:
exactly : `from` default value are now()-1month, `until` default value are now()+1year. It seems suffisent in most of the case to disable off-day in the datepicker UI component.